src/main/java/it/pagopa/interop/probing/statistics/api/service/impl/TimestreamServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package it.pagopa.interop.probing.statistics.api.service.impl;
+
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.CaseFormat;
+import it.pagopa.interop.probing.statistics.api.service.TimestreamService;
+import it.pagopa.interop.probing.statistics.api.util.logging.DateUtilities;
+import it.pagopa.interop.probing.statistics.api.util.logging.Logger;
+import it.pagopa.interop.probing.statistics.dtos.StatisticContent;
+import software.amazon.awssdk.services.timestreamquery.TimestreamQueryClient;
+import software.amazon.awssdk.services.timestreamquery.model.ColumnInfo;
+import software.amazon.awssdk.services.timestreamquery.model.Datum;
+import software.amazon.awssdk.services.timestreamquery.model.QueryRequest;
+import software.amazon.awssdk.services.timestreamquery.model.QueryResponse;
+import software.amazon.awssdk.services.timestreamquery.model.QueryStatus;
+import software.amazon.awssdk.services.timestreamquery.model.Row;
+import software.amazon.awssdk.services.timestreamquery.paginators.QueryIterable;
+
+@Service
+public class TimestreamServiceImpl implements TimestreamService {
+
+  @Autowired
+  private TimestreamQueryClient queryClient;
+
+  @Autowired
+  private ObjectMapper mapper;
+
+  @Autowired
+  private DateUtilities dateUtilities;
+
+  @Value("${amazon.timestream.database}")
+  private String database;
+
+  @Value("${amazon.timestream.table}")
+  private String table;
+
+  private static final String TIME_MEASURE = "time";
+
+  @Autowired
+  private Logger logger;
+
+  @Override
+  public List<StatisticContent> findStatistics(Long eserviceRecordId, Integer pollingFrequency)
+      throws JsonMappingException, JsonProcessingException, ParseException {
```

Comment:
Please use IOException if you don't need to manage all these exceptions

Reply:
ok